skyreader/result.py

```diff
@@ -825,13 +825,41 @@ def create_plot(self, dozoom = False):
 
         print("done.")
 
+    def circular_contour(self, ra, dec, sigma, nside):
+            """For plotting circular contours on skymaps ra, dec, sigma all
+            expected in radians."""
+            dec = np.pi/2. - dec
+            sigma = np.rad2deg(sigma)
+            delta, step, bins = 0, 0, 0
+            delta= sigma/180.0*np.pi
+            step = 1./np.sin(delta)/10.
+            bins = int(360./step)
+            Theta = np.zeros(bins+1, dtype=np.double)
+            Phi = np.zeros(bins+1, dtype=np.double)
+            # define the contour
+            for j in range(0,bins) :
+                phi = j*step/180.*np.pi
+                vx = np.cos(phi)*np.sin(ra)*np.sin(delta) + np.cos(ra)*(np.cos(delta)*np.sin(dec) + np.cos(dec)*np.sin(delta)*np.sin(phi))
+                vy = np.cos(delta)*np.sin(dec)*np.sin(ra) + np.sin(delta)*(-np.cos(ra)*np.cos(phi) + np.cos(dec)*np.sin(ra)*np.sin(phi))
+                vz = np.cos(dec)*np.cos(delta) - np.sin(dec)*np.sin(delta)*np.sin(phi)
+                idx = healpy.vec2pix(nside, vx, vy, vz)
+                DEC, RA = healpy.pix2ang(nside, idx)
+                Theta[j] = DEC
+                Phi[j] = RA
+            Theta[bins] = Theta[0]
+            Phi[bins] = Phi[0]
+            return Theta, Phi
+
     def create_plot_zoomed(self,
                            extra_ra=np.nan,
                            extra_dec=np.nan,
                            extra_radius=np.nan,
                            systematics=False,
                            plot_bounding_box=False,
-                           plot_4fgl=False):
+                           plot_4fgl=False,
+                           circular=False,
+                           circular_err50=0.2,
+                           circular_err90=0.7):
```

Comment:
While this is acceptable to me at this stage, I don't like very much the idea of an ever-growing list of (boolean) arguments in order to control the variant of the plotting.

This leads to a situation where we have 2^n combinations of settings and most of those are not meaningful or have undefined behaviour.

For example, passing `circular=True` would make the value of `systematics` irrelevant.

At some point we may want to define the plotting variants in a more structured way (as part of #22 ).

Reply:
I think it would be nice to keep the option to have both the online splinempe circular contours and other circularized errors at the same time, so I would keep all the arguments. I can add the raising of an error if err90 is smaller than err50.

Reply:
Yes, there are several scenario we should cover, I tried to put together a list in #26 

The existing extra* arguments allow for a single radius at an arbitrary direction, while here we want two radii at the best fit direction.

I am not sure if we should put effort into merging them now considering that all would have to be refactored later.

I guess ultimately we want to achieve something of the kind:
```
def create_plot_zoomed(self, contours: list[ContourModel], catalogs: list[Catalog])
```

Reply:
Thanks @mlincett for creating the issue. I'm fine with merging this in as it sounds like we're all onboard with improving things later.